test/quantization/pt2e/test_quantize_pt2e_qat.py

```diff
@@ -159,8 +159,8 @@ def _verify_symmetric_xnnpack_qat_numerics_helper(
         self.assertEqual(after_prepare_result_pt2e, after_prepare_result_fx)
 
         if verify_convert:
-            torch.ao.quantization.move_exported_model_to_eval(model_pt2e)
             model_pt2e = convert_pt2e(model_pt2e)
+            torch.ao.quantization.move_exported_model_to_eval(model_pt2e)
```

Comment:
Why this change?

Reply:
Previously, `torch.ao.quantization.move_exported_model_to_eval(model_pt2e)` is after `prepare_qat_pt2e` but before `convert_pt2e` in this case. And `convert_pt2e` will do `_fold_conv_bn_qat` which generate `bn` pattern in training mode. So, after we enable `bn` in `move_exported_model_to_eval`, it will fail the conv-bn folding. 
I think we can move it after `convert_pt2e` to make the `_fold_conv_bn_qat` works.

Reply:
> If that's the case we don't even need move_exported_model_to_eval to support batch norm for the conv-bn case, since there's no batch norm op anymore after folding.

Yes, I think so. If all BN folded into Conv, we don't need to call `move_exported_model_to_eval` which is only needed for single BN case.

> Doesn't the bn pattern disappear after calling _fold_conv_bn_qat

I think here, in the previous code,  `move_exported_model_to_eval` is called before `_fold_conv_bn_qat` which fails this `_fold_conv_bn_qat`.

Reply:
Wait, this is actually a problem then. What was the failure? We don't want to impose any ordering requirements between `move_exported_model_to_eval` and `convert_pt2e`. If the test failed in the previous order then we need to fix this

Reply:
> What was the failure? 

The checks for `quant_result_pt2e` and `quant_result_fx` will fail in: https://github.com/pytorch/pytorch/blob/235eaabfedbecabd0c5e05ba5a6d92ba578453b1/test/quantization/pt2e/test_quantize_pt2e_qat.py#L171 I think it's due to `_fold_conv_bn_qat_helper` assume BN is in training mode and will fail to do folding after we enabling `_replace_batchnorm_for_eval` in `_move_exported_model_to_eval`.

>  We don't want to impose any ordering requirements between move_exported_model_to_eval and convert_pt2e. If the test failed in the previous order then we need to fix this

ok, then I think may be we can have 2 options:

- Option 1: Enable the Conv-BN folding in `_fold_conv_bn_qat_helper` when BN is in eval mode. Draft a PR for this option: https://github.com/pytorch/pytorch/pull/114927
- Option 2: Makes `_replace_batchnorm_for_eval` optional in `_move_exported_model_to_eval` and makes it default off.

What do you think of it @andrewor14?